numpy/linalg/linalg.py

```diff
@@ -1568,10 +1572,13 @@ def matrix_rank(M, tol=None):
     M = asarray(M)
     if M.ndim < 2:
         return int(not all(M==0))
-    S = svd(M, compute_uv=False)
+    if hermitian:
+        S = eigvalsh(M)
+    else:
+        S = abs(svd(M, compute_uv=False))
```

Comment:
Yes, I think you should have `abs(eigvalsh(X))` on the line above. Why are you changing what happens for `hermitian = False`? Was there a bug?